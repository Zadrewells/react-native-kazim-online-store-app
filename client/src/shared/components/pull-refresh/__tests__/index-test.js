import React from 'react';
import renderer from 'react-test-renderer';

import {
	PullRefresh
} from 'src/shared/components';

import {
	View
} from 'react-native';

jest.useFakeTimers();

describe('<TranslateAndOpacity />', () => {
	const getWrapper = props => renderer.create(
		<PullRefresh
			{
			...props
			}
		/>
	).toJSON();

	const modelI = {
		isRefreshing: false,
		onRefresh: () => {},
		contentView: (
			<View />
		)
	};

	it('renders correctly', () => {
		const tree = getWrapper(modelI);
		expect(tree).toMatchSnapshot();
	});
});
