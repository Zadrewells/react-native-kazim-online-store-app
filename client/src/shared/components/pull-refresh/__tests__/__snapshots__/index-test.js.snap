// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TranslateAndOpacity /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "width": "100%",
      },
    ]
  }
>
  <View
    refreshAnimationHeight={60}
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": 60,
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "transform": Array [
            Object {
              "rotateZ": "0deg",
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "display": "flex",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "marginRight": 0,
              },
            ]
          }
          tag={false}
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="32"
            bbWidth="32"
            color="#5197ff"
            height="32"
            icon="spinner"
            meetOrSlice={0}
            minX={0}
            minY={0}
            primary={true}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                },
                false,
                Object {
                  "flex": 0,
                  "height": 32,
                  "width": 32,
                },
              ]
            }
            tag={false}
            tintColor="#5197ff"
            vbHeight={64}
            vbWidth={64}
            width="32"
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  0,
                  0,
                  0,
                  1,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth="1"
              x={0}
              y={0}
            >
              <RNSVGCircle
                cx="32"
                cy="32"
                fill={null}
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "stroke",
                    "strokeWidth",
                    "strokeDasharray",
                    "strokeDashoffset",
                    "fill",
                  ]
                }
                r="26"
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={
                  Array [
                    0,
                    0.9411764705882353,
                    0.9411764705882353,
                    0.9411764705882353,
                    1,
                  ]
                }
                strokeDasharray={
                  Array [
                    "300",
                    "300",
                  ]
                }
                strokeDashoffset={82}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth="8"
                x={0}
                y={0}
              />
              <RNSVGCircle
                cx="32"
                cy="32"
                fill={null}
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "stroke",
                    "strokeWidth",
                    "strokeDasharray",
                    "strokeDashoffset",
                    "fill",
                  ]
                }
                r="26"
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={
                  Array [
                    0,
                    0.3176470588235294,
                    0.592156862745098,
                    1,
                    1,
                  ]
                }
                strokeDasharray={
                  Array [
                    "128",
                    "128",
                  ]
                }
                strokeDashoffset={82}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth="8"
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
        "zIndex": 1,
      }
    }
  />
</View>
`;
